### PR TITLE
Create kubeconfig for each cluster in reconcile function

### LINKED ISSUE #410: get secret not found issue
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
latest code...
```
        Name:       master-user-data
        Namespace:  openstack-provider-system
  Versions:
    Control Plane:  1.15.0
    Kubelet:        1.15.0
Events:
  Type     Reason            Age                From                Message
  ----     ------            ----               ----                -------
  Warning  FailedSetNodeRef  4s (x11 over 16s)  noderef-controller  failed to retrieve kubeconfig secret for Cluster "test1" in namespace "default": secret not found

o retrieve kubeconfig secret for Cluster "test1" in namespace "default": secret not found
E0726 07:33:58.314933       1 controller.go:217] kubebuilder/controller "msg"="Reconciler error" "error"="failed to retrieve kubeconfig secret for Cluster \"test1\" in namespace \"default\": secret not found"  "controller"="noderef-controller" "request"={"Namespace":"default","Name":"openstack-master-pf9cx"}
I0726 07:34:03.435002       1 noderef_controller.go:84] Reconcile request for Machine "openstack-master-pf9cx" in namespace "default"
E0726 07:34:03.435176       1 noderef_controller.go:135] Failed to assign NodeRef to Machine "default/openstack-master-pf9cx": failed to retrieve kubeconfig secret for Cluster "test1" in namespace "default": secret not found
E0726 07:34:03.435375       1 controller.go:217] kubebuilder/controller "msg"="Reconciler error" "error"="failed to retrieve kubeconfig secret for Cluster \"test1\" in namespace \"default\": secret not found"  "controller"="noderef-controller" "request"={"Namespace":"default","Name":"openstack-master-pf9cx"}
I0726 07:34:13.675540       1 noderef_controller.go:84] Reconcile request for Machine "openstack-master-pf9cx" in namespace "default"
E0726 07:34:13.675677       1 noderef_controller.go:135] Failed to assign NodeRef to Machine "default/openstack-master-pf9cx": failed to retrieve kubeconfig secret for Cluster "test1" in namespace "default": secret not found
E0726 07:34:13.676434       1 controller.go:217] kubebuilder/controller "msg"="Reconciler error" "error"="failed to retrieve kubeconfig secret for Cluster \"test1\" in namespace \"default\": secret not found"  "controller"="noderef-controller" "request"={"Namespace":"default","Name":"openstack-master-pf9cx"}

```

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### LINKED ISSUE #413: Create masterIP in cluster provider spec
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
from here:
https://cluster-api.sigs.k8s.io/common_code/noderef_controller.html
Infrastructure providers can opt-in to use this controller by providing a kubeconfig as a Kubernetes Secret. The secret must be stored in the namespace the Cluster lives in and named as <cluster-name>-kubeconfig, the data should only contain a single key called value and its data should be a valid Kubernetes kubeconfig.

this means the cluster reconcile has to create kubeconfig and also this means cluster has to know all info to build kubeconfig. we solved CA and other issues ,but how to get master ip is unknown, previously this ip is got from machine annotaion (original value through machine spec definition) ,but now we have to create kubeconfig before machine is ready

so conceptually we should either set the previous floating ip value to master machine into cluster spec (for 1 master only) and Loadbalancer address for 2+ master situation, this also make sense because master ip (API endpoint) need store in cluster spec /status instead of in the master machine spec/status

so we need something below and consider to remove 'floating ip' in master later..
```
pkg/apis/openstackproviderconfig/v1alpha1/types.go

@@ -223,6 +223,10 @@ type OpenstackClusterProviderSpec struct {

        // SAKeyPair is the service account key pair.
        SAKeyPair KeyPair `json:"saKeyPair,omitempty"`
+
+       // MasterIP is the ip of the master, if there is Loadbalancer, should be LB
+       // ip and no LB should be floating ip address
+       MasterIP string `json:"masterIP,omitempty"`
```

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #410 
Fixes #413 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/deployer/deployer.go [hunk 1]
```diff
@@ -55,9 +55,18 @@ func (d *Deployer) GetKubeConfig(cluster *clusterv1.Cluster, master *clusterv1.M
...
+		if err != nil {
+			return "", err
+		}
+	} else {
+		// This case means no master created yet, we need get from cluster info anyway
+		if len(config.MasterIP) == 0 {
+			return "", errors.New("MasterIP in cluster spec not set")
+		}
+		ip = config.MasterIP
 	}
```
Comment (sbueringer) [inline pkg/deployer/deployer.go]: We have to merge this with this code here:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/401/files#diff-266c339b51cf18a076cf106526357c53R70-R78

Comment (sbueringer) [reply pkg/deployer/deployer.go]: I guess this would be in the else branch if ManagedAPIServerLoadBalancer is set to `false`

Comment (jichenjc) [reply pkg/deployer/deployer.go]: yes, let's work on that patch first then rebase from there

Comment (sbueringer) [reply pkg/deployer/deployer.go]: Yup. I'll rebase mine

**Pull request merged**